niftilib/nifti1_tool.c

```diff
@@ -3915,10 +3919,11 @@ int disp_raw_data( void * data, int type, int nvals, char space, int newline )
 int clear_float_zeros( char * str )
 {
    char * dp  = strchr(str, '.'), * valp;
+   int    len;
 
    if( !dp ) return 0;      /* nothing to clear */
 
-   size_t len = strlen(dp);
+   len = (int)strlen(dp);
```

Comment:
Why not use `size_t` here?

Sure, it's super-unlikely that any string passed to this function will be longer than INT_MAX, but still... are you trying to save 4 bytes of stack space?

Reply:
The only reason size_t was added was because of warnings, and casting is generally the more simple and less impactful way to do that.  The function is clearing some number of zeros from a maximum 32 character string.  Unnecessarily promoting variables to higher sizes confuses the purposes of the functions.  int isn't better than size_t here, but it would have been the preferred change in the first place.  This is just more in line with how overall changes should be made to this mature (but not perfect) library.  Unless something is actually broken, have as little impact as possible.

Reply:
>The only reason size_t was added was because of warnings

I turned on `-Wshorten-64-to-32` *to find bugs*, not just to shut the compiler up.  This function `clear_float_zeros` is buggy. It does not do what it's supposed to do when the string is huge. The warning found this bug for us.

>Unnecessarily promoting variables to higher sizes...

There is no promotion to higher sizes.  `strlen` itself returns `size_t`.  `strlen` will correctly compute the length of a huge string.  My changing the `len` variable from `int` to `size_t` only makes it match what `strlen` is *already* giving us. Your changing it back to `int` here is discarding 50% of `strlen`'s result.

We can even [compare the assembly](https://godbolt.org/z/Y4Mradjhe) of the `int` vs `size_t` variants. The truncation to int *takes extra work*, on top of just being incorrect.

>Unless something is actually broken, have as little impact as possible.

I understand not wanting to add regressions to a mature codebase. We indeed have to be careful of that. But something *is* actually broken here, albeit not something likely to be encountered in normal use.

The fix (changing `int` to `size_t`) is extremely safe.

Reply:
Thank you for bring this back to `size_t`. I appreciate you being open to argument!